python/tests/graph/test_import.py

```diff
@@ -0,0 +1,230 @@
+                                                           [_tensor_input_name], [_tensor_output_name]))
+        _check_input_novar(gin)
+
+    def test_saved_model_novar(self):
+        with _make_temp_directory() as tmp_dir:
+            saved_model_dir = os.path.join(tmp_dir, 'saved_model')
+
+            def gin_fun(session):
+                _build_saved_model(session, saved_model_dir)
+                # Build the transformer from exported serving model
+                # We are using signatures, thus must provide the keys
+                return TFInputGraph.fromSavedModelWithSignature(saved_model_dir, _serving_tag,
+                                                                _serving_sigdef_key)
+
+            gin = _build_graph_input(gin_fun)
+            _check_input_novar(gin)
+
+    # TODO: we probably do not need this test
+    def test_saved_graph_novar(self):
+        with _make_temp_directory() as tmp_dir:
+            saved_model_dir = os.path.join(tmp_dir, 'saved_model')
+
+            def gin_fun(session):
+                _build_saved_model(session, saved_model_dir)
+                return TFInputGraph.fromGraph(session.graph, session, [_tensor_input_name], [_tensor_output_name])
+
+            gin = _build_graph_input(gin_fun)
+            _check_input_novar(gin)
+
+    def test_checkpoint_sig_var(self):
+        with _make_temp_directory() as tmp_dir:
+            def gin_fun(session):
+                _build_checkpointed_model(session, tmp_dir)
+                return TFInputGraph.fromCheckpointWithSignature(tmp_dir, _serving_sigdef_key)
+
+            gin = _build_graph_input_var(gin_fun)
+            _check_input_novar(gin)
+
+    def test_checkpoint_nosig_var(self):
+        with _make_temp_directory() as tmp_dir:
+            def gin_fun(session):
+                _build_checkpointed_model(session, tmp_dir)
+                return TFInputGraph.fromCheckpoint(tmp_dir,
+                                                   [_tensor_input_name], [_tensor_output_name])
+
+            gin = _build_graph_input_var(gin_fun)
+            _check_input_novar(gin)
+
+    def test_checkpoint_graph_var(self):
+        with _make_temp_directory() as tmp_dir:
+            def gin_fun(session):
+                _build_checkpointed_model(session, tmp_dir)
+                return TFInputGraph.fromGraph(session.graph, session,
+                                              [_tensor_input_name], [_tensor_output_name])
+
+            gin = _build_graph_input_var(gin_fun)
+            _check_input_novar(gin)
+
+
+_serving_tag = "serving_tag"
+_serving_sigdef_key = 'prediction_signature'
+# The name of the input tensor
+_tensor_input_name = "input_tensor"
+# The name of the output tensor (scalar)
+_tensor_output_name = "output_tensor"
+# The name of the variable
+_tensor_var_name = "variable"
+# The size of the input tensor
+_tensor_size = 3
+
+
+def _build_checkpointed_model(session, tmp_dir):
```

Comment:
(Context: design)
We actually **modify** the graph in the session by assigning signatures to it.  
I am just trying to see if this is okay in terms of immutability.